sdk/ai/azopenaiassistants/client_custom_streaming.go

```diff
@@ -0,0 +1,136 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package azopenaiassistants
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+)
+
+// API Reference: https://platform.openai.com/docs/api-reference/assistants-streaming
+
+// CreateThreadAndRunStreamResponse contains the response from [CreateThreadAndRunStream].
+type CreateThreadAndRunStreamResponse struct {
+	// Stream can be used to stream response events.
+	Stream *EventReader[StreamEvent]
```

Comment:
Is the `StreamEvent` intermediate type really necessary? What I was envisioning was a pattern like this.
```go
event, _ := resp.Stream.Read()
switch e := event.(type) {
case *AssistantThread:
    // e is an *AssistantThread, do something with it....
}
```

Reply:
I was thinking that too, but the issue is that `Kind` is not part of the different model objects like AssistantThread, so I'd have to add it. However, those same models are used in other non-streaming situations.